lib/bindings/langs/android/lib/src/main/kotlin/breez_sdk_liquid_notification/job/LnurlPayVerify.kt

```diff
@@ -0,0 +1,105 @@
+package breez_sdk_liquid_notification.job
+
+import android.content.Context
+import breez_sdk_liquid.BindingLiquidSdk
+import breez_sdk_liquid.GetPaymentRequest
+import breez_sdk_liquid.PaymentDetails
+import breez_sdk_liquid.PaymentState
+import breez_sdk_liquid_notification.Constants.DEFAULT_LNURL_PAY_VERIFY_NOTIFICATION_TITLE
+import breez_sdk_liquid_notification.Constants.DEFAULT_LNURL_PAY_VERIFY_NOTIFICATION_FAILURE_TITLE
+import breez_sdk_liquid_notification.Constants.LNURL_PAY_VERIFY_NOTIFICATION_TITLE
+import breez_sdk_liquid_notification.Constants.LNURL_PAY_VERIFY_NOTIFICATION_FAILURE_TITLE
+import breez_sdk_liquid_notification.Constants.NOTIFICATION_CHANNEL_REPLACEABLE
+import breez_sdk_liquid_notification.NotificationHelper.Companion.notifyChannel
+import breez_sdk_liquid_notification.ResourceHelper.Companion.getString
+import breez_sdk_liquid_notification.SdkForegroundService
+import breez_sdk_liquid_notification.ServiceLogger
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+
+@Serializable
+data class LnurlVerifyRequest(
+    @SerialName("payment_hash") val paymentHash: String,
+    @SerialName("reply_url") val replyURL: String,
+)
+
+// Serialize the response according to to LUD-21 verify specification:
+// https://github.com/lnurl/luds/blob/luds/21.md
+@Serializable
+data class LnurlPayVerifyResponse(
+    val settled: Boolean,
+    val preimage: String?,
+    val pr: String,
+)
+
+class LnurlPayVerifyJob(
+    private val context: Context,
+    private val fgService: SdkForegroundService,
+    private val payload: String,
+    private val logger: ServiceLogger,
+) : LnurlPayJob {
+    companion object {
+        private const val TAG = "LnurlPayVerifyJob"
+    }
+
+    override fun start(liquidSDK: BindingLiquidSdk) {
+        var request: LnurlVerifyRequest? = null
+        try {
+            request = Json.decodeFromString(LnurlVerifyRequest.serializer(), payload)
+            // Get the payment by payment hash
+            val getPaymentReq = GetPaymentRequest.PaymentHash(request.paymentHash)
+            val payment = liquidSDK.getPayment(getPaymentReq) ?: run {
+                throw InvalidLnurlPayException("Not found")
+            }
+            val response = when (val details = payment.details) {
+                is PaymentDetails.Lightning -> {
+                    // In the case of a Lightning payment, if it's paid via Lightning or MRH,
+                    // we can release the preimage
+                    val settled = when (payment.status) {
+                        PaymentState.COMPLETE, PaymentState.PENDING -> details.claimTxId != null
```

Comment:
I guess this means in case of mrh we only consider complete payments (after confirmation)?

Reply:
This actually doesn't take MRH into consideration at all because there will be no `claim_tx_id`. Need to fix this

Reply:
I made a fix to use the payment `tx_id` to determine if the `claim_tx_id` or `mrh_tx_id` is set. If it is set then it can be settled in Pending or Complete. If the `tx_id` is not set then it's Pending because we have the `lockup_tx_id`.